plugins/woocommerce-blocks/assets/js/blocks-registry/product-collection/register-product-collection.tsx

```diff
@@ -26,17 +27,359 @@ export interface ProductCollectionConfig extends BlockVariation {
+	if (
+		config.attributes?.displayLayout !== undefined &&
+		typeof config.attributes.displayLayout !== 'object'
+	) {
+		console.warn(
+			'Invalid displayLayout: displayLayout must be an object.'
+		);
+	}
+	// attributes.collection - It will be derived from args.name
+	if ( config.attributes?.collection !== undefined ) {
+		console.error(
+			'Invalid collection: collection must not be set as it will be automatically derived from config.name.'
+		);
+		return false;
+	}
+	// attributes.hideControls
+	if (
+		config.attributes?.hideControls !== undefined &&
+		! Array.isArray( config.attributes.hideControls )
+	) {
+		console.warn(
+			'Invalid hideControls: hideControls must be an array of strings.'
+		);
+	}
+	// attributes.queryContextIncludes
+	if (
+		config.attributes?.queryContextIncludes !== undefined &&
+		! Array.isArray( config.attributes.queryContextIncludes )
+	) {
+		console.warn(
+			'Invalid queryContextIncludes: queryContextIncludes must be an array of strings.'
+		);
+	}
+
+	/**
+	 * Preview validation
+	 */
+	if ( config.preview !== undefined ) {
+		// preview
+		if ( typeof config.preview !== 'object' || config.preview === null ) {
+			console.warn( 'Invalid preview: preview must be an object.' );
+		}
+		// preview.setPreviewState
+		if (
+			config.preview.setPreviewState !== undefined &&
+			typeof config.preview.setPreviewState !== 'function'
+		) {
+			console.warn(
+				'Invalid preview: setPreviewState must be a function.'
+			);
+		}
+
+		if ( config.preview.initialPreviewState !== undefined ) {
+			// preview.initialPreviewState
+			if ( typeof config.preview.initialPreviewState !== 'object' ) {
+				console.warn(
+					'Invalid preview: initialPreviewState must be an object.'
+				);
+			}
+			// preview.initialPreviewState.isPreview
+			if (
+				typeof config.preview.initialPreviewState.isPreview !==
+				'boolean'
+			) {
+				console.warn(
+					'Invalid preview: preview.isPreview must be a boolean.'
+				);
+			}
+			// preview.initialPreviewState.previewMessage
+			if (
+				typeof config.preview.initialPreviewState.previewMessage !==
+				'string'
+			) {
+				console.warn(
+					'Invalid preview: preview.previewMessage must be a string.'
+				);
+			}
+		}
+	}
+};
+
 /**
  * Register a new collection for the Product Collection block.
  *
  * 🚨🚨🚨 WARNING: This is an experimental API and is subject to change without notice.
  *
- * @param {ProductCollectionConfig} blockVariationArgs The configuration of new collection.
+ * @param {ProductCollectionConfig} config The configuration of new collection.
  */
-export const __experimentalRegisterProductCollection = ( {
-	preview: { setPreviewState, initialPreviewState } = {},
-	...blockVariationArgs
-}: ProductCollectionConfig ) => {
+export const __experimentalRegisterProductCollection = (
+	config: ProductCollectionConfig
+) => {
+	// Check if the config is valid. It will log warnings in the console if the config is invalid.
+	isValidCollectionConfig( config );
```

Comment:
I don't remember if that was the case but if you agree with the bailing early strategy, I think we can also prevent registration if the validation doesn't pass. Something among those lines.

```suggestion
	const isConfigValid = isValidCollectionConfig( config );
	
	if (! isConfigValid) {
	    console.error( `Collection ${config.name} could not be registered due to validation error.` )
	    return false;
	}
```

If you decide to follow that, `isValidCollectionConfig` should return `true` at the end 🙌

Reply:
Makes sense! I will make these changes.